pkg/openstack/actuator.go

```diff
@@ -31,14 +31,20 @@ import (
 	operatorv1 "github.com/openshift/api/operator/v1"
 	log "github.com/sirupsen/logrus"
 
+	"github.com/gophercloud/utils/openstack/clientconfig"
+
 	minterv1 "github.com/openshift/cloud-credential-operator/pkg/apis/cloudcredential/v1"
 	"github.com/openshift/cloud-credential-operator/pkg/operator/constants"
 	actuatoriface "github.com/openshift/cloud-credential-operator/pkg/operator/credentialsrequest/actuator"
 	"github.com/openshift/cloud-credential-operator/pkg/operator/utils"
+
+	yaml "gopkg.in/yaml.v2"
 )
 
 const (
 	rootOpenStackCredsSecretKey = "clouds.yaml"
+	rootOpenStackCloudName      = "openstack"
```

Comment:
We can't assume the cloud entry is named `openstack` - we need to sanitize every cloud entries in the `clouds.yaml` file.

Reply:
In the installer this value is hardcoded, so we always have "openstack" cloud there: https://github.com/openshift/installer/blob/d237b9dee575c6e9ceeba80fd3e137231d907cab/pkg/asset/machines/openstack/machines.go#L29-L30

Reply:
> In the installer this value is hardcoded, so we always have "openstack" cloud there: https://github.com/openshift/installer/blob/d237b9dee575c6e9ceeba80fd3e137231d907cab/pkg/asset/machines/openstack/machines.go#L29-L30

What happens if someone changes that in the installer code?
Sanitizing all entries is not more work and we don't make assumption about the cloud name.

Reply:
@mandre first, why would you want to change the cloud name from "openstack" to something else in the installer code? second, a lot of components assume that the entry is called "openstack" in the clouds.yaml.

CIRO: https://github.com/openshift/cluster-image-registry-operator/blob/master/pkg/storage/swift/swift.go#L113
Manila CSI driver operator: https://github.com/openshift/csi-driver-manila-operator/blob/master/pkg/controllers/secret/secretsync.go#L36-L37
Cinder CSI driver operator: https://github.com/openshift/openstack-cinder-csi-driver-operator/blob/master/assets/configmap.yaml#L7
CCMO: https://github.com/openshift/cluster-cloud-controller-manager-operator/blob/master/pkg/cloud/openstack/assets/config.yaml#L7

Reply:
My point being, hardcoding the cloud name is a code smell.

It doesn't cost us anything to sanitize all entries of the clouds.yaml and it makes the code more robust.